src/Foundation/NSObject2.cs

```diff
@@ -81,6 +81,8 @@ enum Flags : byte {
 			HasManagedRef = 32,
 			// 64, // Used by SoM
 			IsCustomType = 128,
+			// TODO: Which version would generate less code, RequiresUIThread or the negated version?
+			RequiresUIThread = 256,
```

Comment:
How can I check the size increase here?

Reply:
There is probably a lot more API that is thread-safe than not thread-safe, since only these frameworks are considered UI frameworks and get the UI thread check:

https://github.com/xamarin/xamarin-macios/blob/16a84bd7f7abe6d05ca9dc8abd0e97333d14fecf/src/generator.cs#L684-L710

so my guess is that the way you implemented is would be the smallest size increase.